src/frontend/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -107,15 +112,21 @@ export class VideoPlayer extends React.Component<
                 timedTextMode.SUBTITLE,
               ].includes(track.mode),
             )
-            .map(track => (
-              <track
-                key={track.id}
-                src={track.url}
-                srcLang={track.language}
-                kind={trackTextKind[track.mode]}
-                label={track.language}
-              />
-            ))}
+            .map(track => {
+              const language = languageChoices.find(
```

Comment:
Perhaps would be simpler to make this an object outside of the rendered JSX and use `label={languages[track.language] || track.language}`

WDYT?

Reply:
Done